next-app/src/lib/fetch.ts

```diff
@@ -1,40 +1,38 @@
 import { throw_error } from '$lib/error'
 import { start, stop } from '$lib/progress'
+import type { HttpMethod } from '@sveltejs/kit/types/private'
 
-export async function CREATE(url, body) { return await custom_fetch('post'  , url, body) }
-export async function GET   (url      ) { return await custom_fetch('get'   , url      ) }
-export async function UPDATE(url, body) { return await custom_fetch('put'   , url, body) }
-export async function DELETE(url      ) { return await custom_fetch('delete', url      ) }
+type AdaptedFetchArgs = {
+	url: string,
+	method: HttpMethod,
+	body?: object,
+}
+type FetchArgs = {
```

Comment:
thank you, yes, much better:  [3e02ce2](https://github.com/sillsdev/web-languageforge/pull/1664/commits/3e02ce28189ac360f3d6688ba34c71c1bc10a16c)

> I'm intrigued that you decided to start passing an object instead of the url, body & method as individual parameters 😄

partially dogma, I like to strive for "pure" functions if possible.  I also find using these "input" params much more flexible and scalable over time.

Reply:
I personally find this a little easier to read, is this ok?  [c3b7322](https://github.com/sillsdev/web-languageforge/pull/1664/commits/c3b7322281350a3695510d75954792a248c3c434)